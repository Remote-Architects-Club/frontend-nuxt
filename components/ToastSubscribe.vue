<template>
  <div
    v-if="open"
    data-cy="toast-subscribe"
    class="sticky inset-x-0 top-0 flex flex-wrap items-center justify-center py-1 text-white bg-black shadow"
  >
    <p class="mb-2 font-bold text-center sm:mb-0 sm:pr-2">
      Stay in the loop for future updates
      <span class="hidden sm:inline-block">&rarr;</span
      ><span class="sm:hidden">&darr;</span>
    </p>
    <mc-api-form />
    <button
      data-cy="close-btn"
      class="absolute top-0 right-0 p-2 text-lg font-bold text-white"
      @click="open = false"
    >
      &times;
    </button>
  </div>
</template>

<script>
import McApiForm from '@/components/McApiForm.vue'
export default {
  components: {
    McApiForm
  },
  data() {
    return {
      open: true
    }
  }
}
</script>

<style lang="scss" scoped></style>
