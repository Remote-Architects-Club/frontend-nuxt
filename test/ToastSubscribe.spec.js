import { mount, RouterLinkStub } from '@vue/test-utils'
import ToastSubscribe from '@/components/ToastSubscribe.vue'

const factory = () => mount(ToastSubscribe)

describe('ToastSubscribe', () => {
  test('is a Vue instance', () => {
    const wrapper = factory()
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  test('renders properly ', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })
  test('disappears when close button is clicked', () => {
    const wrapper = factory()
    wrapper.find('[data-cy=close-btn]').trigger('click')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
