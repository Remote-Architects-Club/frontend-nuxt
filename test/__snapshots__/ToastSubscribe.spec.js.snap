// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastSubscribe disappears when close button is clicked 1`] = `
"<div class=\\"sticky inset-x-0 top-0 flex flex-wrap items-center justify-center py-1 text-white bg-black shadow\\">
  <p class=\\"mb-2 font-bold text-center sm:mb-0 sm:pr-2\\">
    Stay in the loop for future updates
    <span class=\\"hidden sm:inline-block\\">→</span><span class=\\"sm:hidden\\">↓</span></p>
  <div>
    <form data-cy=\\"subscribe\\" class=\\"flex\\"><input id=\\"email\\" type=\\"email\\" placeholder=\\"Email\\" data-cy=\\"email-input\\" class=\\"p-1 mr-1 text-black transition duration-150 ease-in-out border border-black rounded-none shadow-inner focus:outline-none focus:shadow-focus\\"><button type=\\"submit\\" class=\\"flex-none btn btn-regular\\">
        subscribe
      </button></form>
  </div> <button data-cy=\\"close-btn\\" class=\\"absolute top-0 right-0 p-2 text-lg font-bold text-white\\">
    ×
  </button>
</div>"
`;

exports[`ToastSubscribe renders properly  1`] = `
"<div class=\\"sticky inset-x-0 top-0 flex flex-wrap items-center justify-center py-1 text-white bg-black shadow\\">
  <p class=\\"mb-2 font-bold text-center sm:mb-0 sm:pr-2\\">
    Stay in the loop for future updates
    <span class=\\"hidden sm:inline-block\\">→</span><span class=\\"sm:hidden\\">↓</span></p>
  <div>
    <form data-cy=\\"subscribe\\" class=\\"flex\\"><input id=\\"email\\" type=\\"email\\" placeholder=\\"Email\\" data-cy=\\"email-input\\" class=\\"p-1 mr-1 text-black transition duration-150 ease-in-out border border-black rounded-none shadow-inner focus:outline-none focus:shadow-focus\\"><button type=\\"submit\\" class=\\"flex-none btn btn-regular\\">
        subscribe
      </button></form>
  </div> <button data-cy=\\"close-btn\\" class=\\"absolute top-0 right-0 p-2 text-lg font-bold text-white\\">
    ×
  </button>
</div>"
`;
